core/src/main/java/hudson/PluginWrapper.java

```diff
@@ -574,6 +588,15 @@ public boolean hasLicensesXml() {
                     dependencyErrors.add(Messages.PluginWrapper_obsoleteCore(Jenkins.getVersion().toString(), requiredCoreVersion));
                 }
             }
+
+            String requiredJavaVersion = getRequiredJavaVersion();
+            if (requiredJavaVersion != null) {
+                // TODO replace with calls to Runtime.version() once we're on Java 9
+                VersionNumber actualVersion = new VersionNumber(System.getProperty("java.version"));
```

Comment:
You want `java.specification.version`. The format of `java.version` is undefined.